front/src/applications/operationalStudies/hooks/useAutoUpdateProjection.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect } from 'react';
+
+import { useSelector } from 'react-redux';
+
+import type { InfraWithState } from 'common/api/osrdEditoastApi';
+import type { TrainScheduleWithDetails } from 'modules/trainschedule/components/Timetable/types';
+import {
+  updateSelectedTrainId,
+  updateTrainIdUsedForProjection,
+} from 'reducers/osrdsimulation/actions';
+import { getSelectedTrainId, getTrainIdUsedForProjection } from 'reducers/osrdsimulation/selectors';
+import { useAppDispatch } from 'store';
+
+const useAutoUpdateProjection = (
+  infra: InfraWithState,
+  trainIds: number[],
+  trainScheduleSummaries: TrainScheduleWithDetails[]
+) => {
+  const dispatch = useAppDispatch();
+  const currentProjection = useSelector(getTrainIdUsedForProjection);
```

Comment:
it’s an id like `selectTrainId`

```suggestion
  const projectedTrainId = useSelector(getTrainIdUsedForProjection);
```

Reply:
Done !